### PR TITLE
sat_on_jury not being set

### PR DESCRIPTION
**Pull request by philhead1**
### JIRA link (if applicable) ###
https://centralgovernmentcgi.atlassian.net/browse/JM-6946

### Change description ###
Updated confirmJuryAttendance and return jury

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/integration-test/java/uk/gov/hmcts/juror/api/moj/controller/JurorManagementControllerITest.java [hunk 1]
```diff
@@ -1788,6 +1788,7 @@ void confirmAttendanceHappy() {
             assertThat(appearance.getTimeOut()).isEqualTo(LocalTime.of(17, 00));
             assertThat(appearance.getAppearanceStage()).isEqualTo(EXPENSE_ENTERED);
             assertThat(appearance.getAttendanceAuditNumber()).isEqualTo("J00123456");
+            assertThat(appearance.getSatOnJury()).isTrue();
```
Comment (CCHAPMAN01) [inline src/integration-test/java/uk/gov/hmcts/juror/api/moj/controller/JurorManagementControllerITest.java:1791]: can you update the other tests to make sure the property is not true (validate the negative scenarios)

File: src/test/java/uk/gov/hmcts/juror/api/moj/service/trial/TrialServiceImplTest.java [hunk 2]
```diff
@@ -307,15 +308,12 @@ void testReturnJuryConfirmAttendance(String checkInTime) {
...
             createReturnJuryDto(false, "09:00", "10:00"));
 
+        ArgumentCaptor<Appearance> appearanceArgumentCaptor = ArgumentCaptor.forClass(Appearance.class);
+
         verify(panelRepository, times(1))
             .findByTrialTrialNumberAndTrialCourtLocationLocCode(trialNumber, "415");
         verify(panelRepository, times(panelMembers.size())).saveAndFlush(any());
         verify(jurorHistoryRepository, times(panelMembers.size())).save(any());
-        if ("null".equals(checkInTime)) {
-            verify(appearanceRepository, times(panelMembers.size())).saveAndFlush(any());
```
Comment (CCHAPMAN01) [inline src/test/java/uk/gov/hmcts/juror/api/moj/service/trial/TrialServiceImplTest.java:315]: should keep this line to verify the repo was always called?

File: src/test/java/uk/gov/hmcts/juror/api/moj/service/trial/TrialServiceImplTest.java [hunk 3]
```diff
@@ -307,15 +308,12 @@ void testReturnJuryConfirmAttendance(String checkInTime) {
         trialService.returnJury(payload, trialNumber, "415",
             createReturnJuryDto(false, "09:00", "10:00"));
 
+        ArgumentCaptor<Appearance> appearanceArgumentCaptor = ArgumentCaptor.forClass(Appearance.class);
+
```
Comment (CCHAPMAN01) [inline src/test/java/uk/gov/hmcts/juror/api/moj/service/trial/TrialServiceImplTest.java:312]: doesn't look like this has been used in this method - if you re-instate the below verify() call you can capture and assert the value set or the flag

File: src/test/java/uk/gov/hmcts/juror/api/moj/service/trial/TrialServiceImplTest.java [hunk 4]
```diff
@@ -390,7 +392,9 @@ void testReturnJuryConfirmAttendanceCompleteService() {
         verify(panelRepository, times(panelMembers.size())).saveAndFlush(any());
         verify(completeService, times(panelMembers.size())).completeService(anyString(), any());
         verify(jurorHistoryRepository, times(panelMembers.size())).save(any());
-        verify(appearanceRepository, times(panelMembers.size())).saveAndFlush(any());
+        ArgumentCaptor<Appearance> appearanceArgumentCaptor = ArgumentCaptor.forClass(Appearance.class);
+
+        verify(appearanceRepository, times(panelMembers.size())).saveAndFlush(appearanceArgumentCaptor.capture());
```
Comment (CCHAPMAN01) [inline src/test/java/uk/gov/hmcts/juror/api/moj/service/trial/TrialServiceImplTest.java:397]: no assertion following argument capture

File: src/integration-test/java/uk/gov/hmcts/juror/api/moj/controller/JurorManagementControllerITest.java (hunk 1 above)
Comment (philhead1) [reply src/integration-test/java/uk/gov/hmcts/juror/api/moj/controller/JurorManagementControllerITest.java:1826]: 👍

File: src/integration-test/java/uk/gov/hmcts/juror/api/moj/controller/JurorManagementControllerITest.java [hunk 5]
```diff
@@ -1811,7 +1846,6 @@ void confirmAttendanceHappy() {
                     && jh.getOtherInformation().equalsIgnoreCase("J00123456"))).isTrue();
         }
 
-        @Test
```
Comment (CCHAPMAN01) [inline src/integration-test/java/uk/gov/hmcts/juror/api/moj/controller/JurorManagementControllerITest.java:1811]: why has this been removed?

**Review by CCHAPMAN01 — APPROVED** [review]

**Pull request merged**